legend-depot-store-metrics/src/main/java/org/finos/legend/depot/store/metrics/api/QueryMetricsRegistry.java

```diff
@@ -0,0 +1,31 @@
+//  Copyright 2021 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+package org.finos.legend.depot.store.metrics.api;
+
+import org.finos.legend.depot.store.admin.domain.metrics.VersionQueryMetric;
+
+import java.util.Date;
+
+public interface QueryMetricsRegistry
+{
+    void record(String groupId, String artifactId, String versionId);
+
+    void record(String groupId, String artifactId, String versionId, Date date);
+
+    VersionQueryMetric getFirstInQueue();
```

Comment:
why do we need get first and find first  in queue... u are exposing your implementation
the methods can be consolidated into one methof findFirst which returns an optional and u deal with that from where u called it